integrations/clients/hud_income_limits/client.py

```
@@ -279,6 +305,64 @@ def _normalize_fmr_basicdata(basic_data: object, bedroom_field: str) -> dict:
                     return item
         return {}
 
+    def _validate_bedrooms(self, bedrooms: int) -> None:
+        if bedrooms < 0 or bedrooms > 4:
+            raise HudIncomeClientError(f"Bedroom count must be 0–4, got {bedrooms}")
```

Ruff (0.15.20)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)